pep-XXXX.rst

```diff
@@ -0,0 +1,492 @@
+Post-History:
+
+Abstract
+========
+
+This PEP introduces a concise and friendly syntax for callable types, supporting the same functionality as ``typing.Callable`` but with an arrow syntax inspired by the syntax for typed function signatures. This allows types like ``Callable[[int, str], bool]`` to be written ``(int, str) -> bool``.
+
+The proposed syntax supports all the functionality provided by ``typing.Callable`` and ``typing.Concatenate``, and is intended to work as a drop-in replacement.
+
+
+Motivation
+==========
+
+
+The ``Callable`` type, defined as part of PEP 484, is one of the most commonly used complex types in ``typing`` alongside ``Union`` and collection types like ``Dict`` and ``List``.
+
+
+There are four major problems with the existing ``Callable`` type:
+- It is verbose, particularly for more complex function signatures.
+- It requires an explicit import, something we no longer require for most of the other
+  very common types after PEP 604 (``|`` for ``Union`` types) and PEP 585 (generic collections)
+- It does not visually represent the way function headers are written.
+- It relies on two levels of nested square brackets. This can be quite hard to read,
+  especially when the function arguments themselves have square brackets.
+
+It is common for library authors to make use of untyped or partially-typed callables (e.g. ``Callable[..., Any]``) which we believe is partially a result of the existing types being hard to use. Libraries with less precise types reduce the ability of static analyzers running on downstream projects (including type checkers and security analysis tools) to find problems.
+
+With a succinct, easy-to-use syntax, developers may be less likely to reach for poorly-typed options. Callable types may also be beginner-friendly if we make them look more like function headers, and like the arrow type syntax used by several other popular languages.
+
+A simplified real-world example from a web server illustrates how the types can be verbose and require many levels of nested square brackets::
+
+    from typing import Awaitable, Callable
+    from app_logic import Response, UserSetting
+
+
+    async def customize_response_for_settings(
+        response: Response,
+        customizer: Callable[[Response, list[UserSetting]], Awaitable[Response]]
+    ) -> Response:
+       ...
+
+With our proposal, this code can be abbreviated to::
+
+    from app_logic import Response, UserSetting
+
+    def make_endpoint(
+        response: Response,
+        customizer: async (Response, list[UserSetting]) -> Response,
+    ) -> Response:
+        ...
+
+This is shorter and requires fewer imports. It also has far less nesting of square brackets - only one level, as opposed to three in the original code.
+
+Rationale
+=========
+
+The ``Callable`` type is widely used. For example, in typeshed [#typeshed-stats]_ it is the fifth most common complex type, after ``Optional``, ``Tuple``, ``Union``, and ``List``.
```

Comment:
Would it be worth noting that this statistic is based on analysis done in October 2021? (Things might always change! The heavy presence of `Optional`, `Tuple`, `Union` and `List` in typeshed is due to a bug in mypy that means PEP 585 and PEP 604 can't yet be used in all situations. But a new release of mypy is just around the corner...)

Reply:
Good point. I think using current stats is most fair since we're arguing by analogy here, but we should be specific about the time